### DISCUSSION TITLE
SPM Release/Debug?

### DISCUSSION THREAD

**Discussion by AndrewSB** [pointfreeco/isowords#92] [Q&A] [answered]
hey! i have a similar structure, and i was wondering if you two ever had issues with your packages running in `RELEASE` mode while you're debugging? this is something i've noticed more recently (so it might be related to the Xcode 12.5 beta).

sometimes, i notice that even though my app scheme is Debug, and all of my autogenned Package schemes say `DEBUG` as well, my packages are being run in `RELEASE`, while the app is in `DEBUG`. My release/debug distinction here is whether the `#if DEBUG` flag is set, or if `assertionFailure` and friends will result in a trap

thanks for open sourcing! i've love learning from you 🙃

**Comment by stephencelis** [comment, marked as answer]
We haven't noticed this for `DEBUG` vs. `RELEASE`, but there is a known bug that custom schemes are treated as `RELEASE`. If you encounter it while debugging isowords and have a good way of reproducing it, let us know!

**Reply by AndrewSB** [reply]
[no response needed, leaving here for posterity]

I've noticed that this behavior happens if the project/workspace build configuration is named anything but `debug`. I had renamed my configuration to `Staging`, switching back to `Debug` fixed it for me. seems like an Xcode bug

<img width="632" alt="Screen Shot 2021-04-24 at 11 36 32 AM" src="https://user-images.githubusercontent.com/3814772/115969341-52e8e380-a4f1-11eb-882d-3c874e6dba56.png">
